jib-core/src/main/java/com/google/cloud/tools/jib/ncache/DefaultCacheStorageFiles.java

```diff
@@ -18,15 +18,53 @@
 
 import com.google.cloud.tools.jib.image.DescriptorDigest;
 import java.nio.file.Path;
+import java.security.DigestException;
 
 /** Resolves the files used in the default cache storage engine. */
 class DefaultCacheStorageFiles {
 
   private static final String LAYERS_DIRECTORY = "layers";
-  private static final String LAYER_FILENAME_SUFFIX = ".layer";
   private static final String METADATA_FILENAME = "metadata";
   private static final String SELECTORS_DIRECTORY = "selectors";
 
+  /**
+   * Returns whether or not {@code file} is a layer contents file.
+   *
+   * @param file the file to check
+   * @return {@code true} if {@code file} is a layer contents file; {@code false} otherwise
+   */
+  static boolean isLayerFile(Path file) {
+    return file.getFileName().toString().length() == DescriptorDigest.HASH_LENGTH;
```

Comment:
Can/should we do better here? This doesn't seem like a very thorough check. I could see someone setting an arbitrary 64-character filename to get past this check or something.

Reply:
I think the assumptions for this cache is that the user will not be messing with its contents, and if they do, we have no guarantees about possible data corruption, so such a check here I think would be okay rather than fully matching against the hash regex. For example, the user could modify the layer file itself and produce completely corrupted layer contents, but we would not be liable for detecting/fixing that.

Reply:
Maybe we should at least verify the layer contents still hash to the same value?

Reply:
Yea, the hashing is an expensive operation. I believe that even if we had that check, since the selector file could be modified as well, a selector generated from a set of layer entries could select a completely different layer than intended and we don't have an easy way of checking the validity of this with anything less than having to decompress the layer and read all the tar archive entries.

Reply:
Oh it's a hash of the decompressed file? Can we verify integrity with a hash of the compressed file somewhere?

Reply:
If the compressed hash doesn't actually match the uncompressed hash, the Docker registry would actually reject it (hence why they have diff ID in addition to the layer digest), though the corruption would be hidden under a manifest invalid error from the registry during push. I'll file a separate issue to explore potential cache corruptions to detect and test for solutions that may be adequate and fast.

Reply:
:+1: